spec/features/hiring_staff_can_specify_a_part_time_role.rb

```diff
@@ -23,18 +23,65 @@
   end
 
   scenario 'the list page should indicate a pro rata salary if the role is part time', elasticsearch: true do
-    create(:vacancy, working_pattern: :full_time)
     create(:vacancy, working_pattern: :part_time)
 
     Vacancy.__elasticsearch__.client.indices.flush
 
     visit jobs_path
 
-    expect(page.find('.vacancy:eq(2)')).to have_content('Full time')
-    expect(page.find('.vacancy:eq(2)')).to have_content(I18n.t('jobs.per_year'))
-    expect(page.find('.vacancy:eq(2)')).not_to have_content(I18n.t('jobs.per_year_pro_rata'))
-
     expect(page.find('.vacancy:eq(1)')).to have_content('Part time')
     expect(page.find('.vacancy:eq(1)')).to have_content(I18n.t('jobs.per_year_pro_rata'))
   end
+
+  scenario 'the list page should not indicate a pro rata salary if the role is full time', elasticsearch: true do
+    create(:vacancy, working_pattern: :full_time)
+
+    Vacancy.__elasticsearch__.client.indices.flush
+
+    visit jobs_path
+
+    expect(page.find('.vacancy:eq(1)')).to have_content('Full time')
+    expect(page.find('.vacancy:eq(1)')).to have_content(I18n.t('jobs.per_year'))
+    expect(page.find('.vacancy:eq(1)')).not_to have_content(I18n.t('jobs.per_year_pro_rata'))
+  end
+
+  scenario 'a job page should indicate a pro rata salary if the role is part time' do
+    school = create(:school)
+    vacancy = create(:vacancy, :published, school: school, working_pattern: :part_time)
+
+    visit job_path(vacancy.id)
+
+    within('.vacancy--metadata') { expect(page).to have_content('pro rata') }
+  end
+
+  scenario 'a job page should not indicate a pro rata salary if the role is full time' do
+    school = create(:school)
+    vacancy = create(:vacancy, :published, school: school, working_pattern: :full_time)
+
+    visit job_path(vacancy.id)
+
+    within('.vacancy--metadata') { expect(page).not_to have_content('pro rata') }
+  end
+
+  context 'when the user is on mobile' do
```

Comment:
This is complete coverage. It adds 5 tests to ensure pro rata is always shown and it should catch all regressions in the future which is valuable.

I would use these features to drive the development and then look to slim them down by moving some to unit tests, while keeping my confidence high that we'd have a strong change of knowing if it broke in future.

I'd think about updating the [happy path test](https://github.com/dxw/teacher-vacancy-service/blob/develop/spec/features/hiring_staff_can_publish_a_vacancy_spec.rb#L75) to also test for the presence of the correct `pro rota` value. Once here we can see how we're verifying all the other data we add is set as we expect https://github.com/dxw/teacher-vacancy-service/blob/develop/spec/support/vacancy_helpers.rb#L41 and add our new one on. You can see that there's a big reliance on using the `VacancyPresenter` for this purpose, which is another benefit of moving the logic out of the view.

For the rest I'd keep a feature test for what happens when the vacancy is `part_time` which keeps it inline with the spec file is named.

This leaves the mobile tests. Would it be an idea to have a deliberate test for this `_job_seekers_can_see_listings_on_mobile` which tests this and more to make sure that any changes made to show.html+phone.haml are caught.